sha256.c

```diff
@@ -149,52 +149,37 @@ static void sha256_init(struct sha256_state *sctx)
 	};
 }
 
-static void sha256_update(struct sha256_state *sctx, const void *data, u32 len)
+static void sha256_once(struct sha256_state *sctx, const void *data, u32 len)
```

Comment:
So it took me a while to get this but I see what you did not. One thought that may not be worth much. Do you even need the separate *_once() functions? It seems it could all just be 1 sha1 and one sha256 function. Not sure if that would save a bit of space...

Reply:
Actually, looking at it a second time, I am wondering the same. Especially for how the hashing is used in the LZ. The reason for the _update is to handle streaming data. For the LZ everything is fixed blocks, so being able to handle unknown sized data blocks is not really necessary.

Reply:
Observe from the compiled output that sha{1,256}_once() aren't present.  They've been inlined due to having a single caller.

However, keeping the function separate is still good for a code clarity point of view

Reply:
Fair enough.

Reply:
Sounds good to me